bootcompiler/src/main/scala/org/mozartoz/bootcompiler/fastparse/TypeAST.scala

```diff
@@ -58,9 +58,9 @@ object TypeAST {
     case LockObjectPhrase(body) =>
       treeCopy.LockObjectExpression(p, expr(body))
     case ProcPhrase(name, args, body, flags) =>
-      treeCopy.ProcExpression(p, name, args, stat(body), flags)
+      treeCopy.ProcExpression(p, Some(RawVariable(name)), args, stat(body), flags)
```

Comment:
This should not be the job of TypeAST :p
How about typing ProcPhrase with Option[VariableOrRaw] and doing this in the parser?